nsflow/backend/utils/logutils/process_log_bridge.py

```diff
@@ -0,0 +1,462 @@
+            return True
+        if '"request_id"' in line and line.rstrip().endswith("}"):
+            return True
+        return False
+
+    @staticmethod
+    def _reasm_flush(state: Dict[str, Any]) -> str:
+        text = "\n".join(state["buffer"]).strip()
+        state["buffer"].clear()
+        state["balance"] = 0
+        state["collecting"] = False
+        return text
+
+    # ---------- severity ----------
+    def _infer_level_from_message_type(self, record: Dict[str, Any]) -> int:
+        mt = str(record.get("message_type", "")).strip().lower()
+        return self._MESSAGE_TYPE_TO_LEVEL.get(mt, logging.INFO)
+
+    def _infer_level_from_text(self, line: str, default: int = logging.INFO) -> int:
+        if not line:
+            return default
+        m = self._LEVEL_WORD.search(line)
+        if not m:
+            if "traceback" in line.lower():
+                return logging.ERROR
+            return default
+        word = m.group(1).upper()
+        return logging.CRITICAL if word == "FATAL" else getattr(logging, word, default)
+
+    # ---------- json helpers ----------
+    @staticmethod
+    def _pretty_json(obj: Any) -> str:
+        try:
+            return json.dumps(obj, indent=2, ensure_ascii=False)
+        except Exception:
+            return str(obj)
+
+    @staticmethod
+    def _try_parse_json_fragment(text: str) -> Optional[Dict[str, Any]]:
+        if not text:
+            return None
+        # strict
+        try:
+            obj = json.loads(text)
+            return obj if isinstance(obj, dict) else {"message": obj}
+        except Exception:
+            pass
+        # first {...}
+        s = text.find("{")
+        e = text.rfind("}")
+        if s != -1 and e != -1 and e > s:
+            frag = text[s : e + 1]
+            try:
+                obj = json.loads(frag)
+                return obj if isinstance(obj, dict) else {"message": obj}
+            except Exception:
+                return None
+        return None
+
+    @staticmethod
+    def _lenient_inner_json_parse(val: Any) -> Optional[Any]:
+        if not isinstance(val, str):
+            return None
+        s = val.strip()
+        if not s:
+            return None
+        if not (s.lstrip().startswith("{") or s.lstrip().startswith("[")):
+            return None
+        # strict
+        try:
+            return json.loads(s)
+        except Exception:
+            pass
+        # mild cleanup: unescape \n \t \r and drop trailing commas
+        s2 = s.replace("\\r", "\r").replace("\\t", "\t").replace("\\n", "\n")
+        s2 = re.sub(r",\s*(?=[}\]])", "", s2)
+        s2 = re.sub(r"\n{3,}", "\n\n", s2)
+        try:
+            return json.loads(s2)
+        except Exception:
+            return None
+
+    # ---------- special NeuroSan block ----------
+    def _rebuild_neurosan_request_reporting(self, text_block: str) -> Optional[Dict[str, Any]]:
+        m = self._REQUEST_REPORTING_INNER.search(text_block)
```

Comment:
A docstring would help understand what's special for NeuroSan and what needs to be done